internal/mode/static/state/change_processor.go

```diff
@@ -53,12 +53,14 @@ type ChangeProcessorConfig struct {
 	RelationshipCapturer relationship.Capturer
 	// Validators validate resources according to data-plane specific rules.
 	Validators validation.Validators
-	// Logger is the logger for this Change Processor.
-	Logger logr.Logger
 	// EventRecorder records events for Kubernetes resources.
 	EventRecorder record.EventRecorder
 	// Scheme is the Kubernetes scheme.
 	Scheme *runtime.Scheme
+	// ProtectedPorts are the ports that may not be configured by a listener.
+	ProtectedPorts map[int32]string
```

Comment:
can we also document what the map holds as `string` ?

I wonder if should have a type aliases like `type ProtectedPorts map[int32]string` for which we can document what the map holds.
If you look into other places in the code, just by looking at protectedPorts map[int32]string as in func `createPortConflictResolver(protectedPorts map[int32]string) listenerConflictResolver {`, it is hard to tell what the map holds.